src/main/java/br/com/labbs/monitor/filter/MetricsCollectorFilter.java

```diff
@@ -80,49 +93,74 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
-		final String method = httpRequest.getMethod();
-		final String statusRange = Integer.toString(counterResponse.getStatus());
-		final long count = counterResponse.getByteCount();
-		//TODO why do not use a status code range instead of the specific one?
-//            final String statusRange = counterResponse.getStatusRange();
-		DebugUtil.debug(path, " ; bytes count = ", count);
-		MonitorMetrics.INSTANCE.requestSeconds.labels(httpRequest.getScheme(), statusRange, method, path)
-				.observe(elapsedSeconds);
-		MonitorMetrics.INSTANCE.responseSize.labels(httpRequest.getScheme(), statusRange, method, path).inc(count);
+        if (isExcludedPath(httpRequest, path)) {
+            chain.doFilter(request, response);
+        } else {
+            final CountingServletResponse counterResponse =
+                    new CountingServletResponse((HttpServletResponse) response);
+            try {
+                chain.doFilter(httpRequest, counterResponse);
+            } finally {
+                collect(httpRequest, counterResponse, path, timer.elapsedSeconds());
+            }
+        }
     }
 
     @Override
     public void destroy() {
         //ignored
     }
 
+    private boolean isExcludedPath(final HttpServletRequest httpRequest, String path) {
+        if (path.startsWith(httpRequest.getContextPath())) {
+            path = path.substring(httpRequest.getContextPath().length());
+        }
+        for (String exclusion : exclusions) {
+            if (path.startsWith(exclusion)) {
+                DebugUtil.debug("Excluded ", path);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void collect(HttpServletRequest httpRequest, CountingServletResponse counterResponse, String path, double elapsedSeconds) throws IOException {
+        final String method = httpRequest.getMethod();
+        final String statusRange = Integer.toString(counterResponse.getStatus());
+        final long count = counterResponse.getByteCount();
+        DebugUtil.debug(path, " ; bytes count = ", count);
+        MonitorMetrics.INSTANCE.requestSeconds.labels(httpRequest.getScheme(), statusRange, method, path)
+                .observe(elapsedSeconds);
+        MonitorMetrics.INSTANCE.responseSize.labels(httpRequest.getScheme(), statusRange, method, path).inc(count);
+    }
+
     private boolean isEmpty(String s) {
-        return s == null || s.length() == 0;
+        return s == null || s.trim().length() == 0;
     }
+
+    private String substringMaxDepth(String path, int pathDepth) {
+        if (path == null || pathDepth < 1) {
+            return path;
+        }
+        if (!path.startsWith("/")) {
+            path = "/" + path;
+        }
+
+        int count = 0;
+        int i = -1;
+        do {
+            i = path.indexOf("/", i + 1);
+            if (i < 0) {
```

Comment:
@werberson, if I am reading your code correctly, `i` here will never be less than `0`, since on line 150 you are guaranteeing the existence of `/` at index `0`.